app/scenes/Document/components/MarkAsViewed.ts

```diff
@@ -4,29 +4,27 @@ import Document from "~/models/Document";
 const MARK_AS_VIEWED_AFTER = 3 * 1000;
 type Props = {
   document: Document;
+  children: JSX.Element;
 };
 
-class MarkAsViewed extends React.Component<Props> {
-  viewTimeout: ReturnType<typeof setTimeout>;
+function MarkAsViewed(props: Props) {
+  const { document, children } = props;
+  const viewTimeout = React.useRef<ReturnType<typeof setInterval> | null>(null);
 
-  componentDidMount() {
-    const { document } = this.props;
-    this.viewTimeout = setTimeout(async () => {
+  React.useEffect(() => {
+    viewTimeout.current = setTimeout(async () => {
       const view = await document.view();
 
       if (view) {
         document.updateLastViewed(view);
       }
     }, MARK_AS_VIEWED_AFTER);
-  }
 
-  componentWillUnmount() {
-    clearTimeout(this.viewTimeout);
-  }
+    return () =>
+      clearTimeout(viewTimeout.current as ReturnType<typeof setInterval>);
```

Comment:
I think you want to check if `viewTimeout.current` exists here rather than casting

Reply:
sure!